packages/services/src/services/VideoPlayer/__tests__/VideoPlayer.test.js

```diff
@@ -0,0 +1,46 @@
+import apiDataResponse from './data/response.json';
+import root from 'window-or-global';
+import VideoPlayerAPI from '../VideoPlayer';
+
+//jest.mock('../../VideoPlayer');
+
+describe('VideoPlayer', () => {
+  /**
+   *
+   * Mock function for kWidget api and embed objects
+   */
+  const mockFn = () => {
+    root.kWidget = {
+      api: {
+        doRequest: jest.fn(() => Promise.resolve(apiDataResponse)),
+      },
+      embed: jest.fn(() =>
+        Promise.resolve({
+          targetId: 'kaltura_player',
+          wid: '_' + 123456,
+          uiconf_id: 12345678,
+          entry_id: '0_uka1msg4',
+          flashvars: {
+            autoPlay: false,
+          },
+        })
+      ),
+    };
+  };
+  it('should load script', () => {
+    process.env.script =
+      'https://cdnapisec.kaltura.com/p/243342/sp/24334200/embedIframeJs/uiconf_id/12905712/partner_id/243342';
+    VideoPlayerAPI.checkScript();
+    expect(document.body.innerHTML).toEqual(
+      '<script src="https://cdnapisec.kaltura.com/p/243342/sp/24334200/embedIframeJs/uiconf_id/12905712/partner_id/243342"></script>'
+    );
+  });
+
+  it('should return the apiData', async done => {
+    const data = await VideoPlayerAPI.api('0_uka1msg4');
+    mockFn();
```

Comment:
I see that system check is failing since the await is still waiting for `checkScript` to clear, and the `mockFn` isn't reached yet because of this. I think let's call `mockFn` before the line above instead. We're already checking `checkScript` in the other unit test.

Reply:
okay.

Reply:
getting api is undefined when I am running tests

Reply:
I saw the system check. You might need to add `window-or-global` in the service itself, and call `root.kWidget.api` instead of `kWidget.api`.

Reply:
After changing above getting `TypeError: _windowOrGlobal.default.kWidget.api is not a function` an error.